src/v/config/property.h

```diff
@@ -523,4 +523,75 @@ class deprecated_property : public property<ss::sstring> {
 
     void set_value(std::any) override { return; }
 };
+
+template<typename T>
+class enum_property : public property<T> {
+public:
+    enum_property(
+      config_store& conf,
+      std::string_view name,
+      std::string_view desc,
+      base_property::metadata meta,
+      T def,
+      std::vector<T> values)
+      : property<T>(
+        conf,
+        name,
+        desc,
+        meta,
+        def,
+        [this](T new_value) -> std::optional<ss::sstring> {
+            auto found = std::find_if(
+              _values.begin(), _values.end(), [&new_value](T const& v) {
+                  return v == new_value;
+              });
+            if (found == _values.end()) {
+                return help_text();
+
```

Comment:
nit: extra empty line?

Reply:
hmm, yeh it is a little odd. presumably clang-format is ok with it?

Reply:
oops, this is an example of me very much relying on clang-format to do my whitespace for me.  Fixed.